ecommers/lib/core/models/data_models/product.dart

```diff
@@ -47,4 +47,20 @@ class Product {
       _$ProductFromJson(json);
 
   Map<String, dynamic> toJson() => _$ProductToJson(this);
+
+  Map<String, dynamic> toSpecifiedJson() => <String, dynamic>{
```

Comment:
because to save data contains types which are not serialized and to save data in DB we should make deep serialization, because 'toJson' doesn't do it by default.

Reply:
what do u mean by 'deep serialization'? I've checked default toJson() method, and for me, it works fine.

Reply:
Seems, i need show a demo  :)